src/Administration/Service/AdminSearcher.php

```diff
@@ -34,7 +39,7 @@ public function search(CriteriaCollection $entities, Context $context): array
             $repository = $this->definitionRegistry->getRepository($entityName);
             $collection = $repository->search($criteria, $context);
 
-            $result[$entityName] = [
+            $result[(string) $entityName] = [
```

Comment:
should we assert it instead of casting?

Reply:
I prefer casting, as assert, to me, is more a test. What do you prefer?

Reply:
Therefore I think the `assert` is the best way, in particular as nothing else can break in that case.

Otherwise one should put that in an if condition and throw an error, to be consistent, as the developer, after the typehint, expects the thing to be a string.

Reply:
I suppose changing the return type makes more sense in this case, since it comes from a `Collection` with this defined as the type already.